### PR TITLE
fix: schema-path handles falsy values and JSON path expression as field

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/spectral/issues/916

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/functions/__tests__/schema-path.test.ts [hunk 1]
```diff
@@ -65,6 +69,56 @@ describe('schema', () => {
...
+    const target = {
+      schema: {
+        type: 'string',
+      },
+      example: null,
+    };
+    expect(runSchemaPath(target, fieldToCheck, path)).toEqual([
+      {
+        path: ['example'],
+        message: '{{property|gravis|append-property|optional-typeof}}type should be string',
```
Comment (philsturgeon) [inline src/functions/__tests__/schema-path.test.ts]: What is this syntax? Does type need a space before it?

File: src/utils/__tests__/getLintTargets.spec.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+      expect(getLintTargets({ a: void 0 }, 'a')).toStrictEqual([
+        {
+          path: ['a'],
+          value: void 0,
+        },
+      ]);
+    });
+
+    it('given non-existing property, returns the whole document', () => {
```
Comment (nulltoken) [inline src/utils/__tests__/getLintTargets.spec.ts]: @P0lip Either the test title is misleading or there's something I don't understand. Where is the "whole document" in the provided output?

Comment (P0lip) [reply src/utils/__tests__/getLintTargets.spec.ts]: ah, I should have stated `input`

Comment (P0lip) [inline src/utils/__tests__/getLintTargets.spec.ts]:
```suggestion
    it('given non-existing property, returns the whole input', () => {
```

File: src/functions/__tests__/schema-path.test.ts (hunk 1 above)
Comment (P0lip) [reply src/functions/__tests__/schema-path.test.ts]: It's replaced at runtime, when Spectral processes messages returned by functions.
`|` acts kind of like the unix pipe.
So in this case, this is what happens:
- `property` prints the property name
- `gravis` is a function that decorates a property with gravises (`\`property\``)
- `append-property` adds `\`${property} property\`` if there is a property or return empty string if there is no property
- optional-typeof prints a type of linted value if `append-property` returned an empty string or the result of append-property.

I know this sounds weird, but so perhaps a note in documentation would be nice to have?

What user receives is something like
```
message: '`example` property type should be number',
```
or
```
message: `number should be equal to one of the allowed values: 1, 3, 5, 10, 12`,
```

That does make sense?

Comment (philsturgeon) [reply src/functions/__tests__/schema-path.test.ts]: If this is something users should know about then please let's document it.

Comment (m-mohr) [reply src/functions/__tests__/schema-path.test.ts]: This also confused me a lot when working on my recent PRs....

Comment (P0lip) [reply src/functions/__tests__/schema-path.test.ts]: > If this is something users should know about then please let's document it.

Will do!

> This also confused me a lot when working on my recent PRs....

My apologies! Going to document that syntax this week.

**Pull request merged**